pkg/routine/routine.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/10/01 by Marius Neugebauer
+
+// Package routine helps in starting background tasks.
+package routine
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/pace/bricks/http/oauth2"
+	"github.com/pace/bricks/maintenance/errors"
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// Forever is so far in the future that it basically never ends.
+var Forever = time.Duration(1<<63 - 1)
+
+// Run runs the given function in a new context with the given timeout. The new
+// context inherits the logger and oauth2 authentication of the passed context.
+// Panics thrown in the function are logged and sent to sentry. The routines
+// context is cancelled if the program receives a shutdown signal (SIGINT,
+// SIGTERM).
+func Run(ctx context.Context, timeout time.Duration, routine func(context.Context)) (cancel context.CancelFunc) {
+	// transfer logger, authentication and error info
+	routineCtx := log.Ctx(ctx).WithContext(context.Background())
```

Comment:
As an idea, if we did have an auto-inc counter we could use that to give each of the routines a unique name and add that to the logger, that way, we can see all the log statements of one routine. the id could also be used for the context map

Reply:
I like the idea. Would definitely make the logs more readable. Will do. But I see no use in using the id in the `contexts` map. It's already a pointer and therefore cheap as an index.